pkg/printers/internalversion/printers.go

```diff
@@ -1155,8 +1165,24 @@ func printJob(obj *batch.Job, options printers.GenerateOptions) ([]metav1.TableR
 	default:
 		jobDuration = duration.HumanDuration(obj.Status.CompletionTime.Sub(obj.Status.StartTime.Time))
 	}
+	var status string
+	if obj.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
This should go after checking Complete or Failed. Otherwise it might look like the job is still running.

Reply:
I had this condition as the first one because checking for "Complete" or "Fail" is done through conditions. So, if the job has already been finished due to being complete or failed, and you mark it for deletion, it will never show the "Terminating" status.

However, the "Terminating" status is also transitive. Is it worth keeping it at all? cc. @mimowo

Reply:
I don't have a strong view, but it can be lengthy if one deletes the job with `kubectl delete job/<job name> --cascade=foreground`. In that case the terminating job needs to wait for all pods to be deleted first. This can take a long time.  OTOH, `foreground` is not the default deletion propagation policy, so typically it is very fast indeed.

Reply:
Done